example/App.tsx

```diff
@@ -128,24 +128,35 @@ export default function App() {
     console.log("[event]: languagedetection", ev);
   });
 
-  const startListening = () => {
+  const startListening = async () => {
     if (status !== "idle") {
       return;
     }
     setTranscription(null);
     setError(null);
     setStatus("starting");
 
-    ExpoSpeechRecognitionModule.requestPermissionsAsync().then((result) => {
-      console.log("Permissions", result);
-      if (!result.granted) {
-        console.log("Permissions not granted", result);
+    const microphonePermissions =
+      await ExpoSpeechRecognitionModule.requestMicrophonePermissionsAsync();
+    console.log("Microphone permissions", microphonePermissions);
+    if (!microphonePermissions.granted) {
+      setError({ error: "not-allowed", message: "Permissions not granted" });
+      setStatus("idle");
+      return;
+    }
+
+    if (!settings.requiresOnDeviceRecognition) {
```

Comment:
```suggestion
    if (!settings.requiresOnDeviceRecognition && Platform.OS === "ios") {
```

Just to be more clear to anyone that's copying this